src/test/lib/HandshakeTest.cpp

```diff
@@ -227,223 +227,220 @@ QuicTestConnect(
-                }
+        if (ServerStatelessRetry) {
+            TEST_TRUE(Client.GetStatistics().StatelessRetry);
+        }
 
-                TEST_EQUAL(
-                    Server->GetPeerBidiStreamCount(),
-                    Client.GetLocalBidiStreamCount());
+        if (SessionResumption == QUIC_TEST_RESUMPTION_ENABLED ||
+            SessionResumption == QUIC_TEST_RESUMPTION_ENABLED_ASYNC) {
+            TEST_TRUE(Client.GetResumed());
+            TEST_TRUE(Server->GetResumed());
+        } else if (SessionResumption == QUIC_TEST_RESUMPTION_REJECTED ||
+                   SessionResumption == QUIC_TEST_RESUMPTION_REJECTED_BY_SERVER_APP ||
+                   SessionResumption == QUIC_TEST_RESUMPTION_REJECTED_BY_SERVER_APP_ASYNC) {
+            TEST_FALSE(Client.GetResumed());
+            TEST_FALSE(Server->GetResumed());
+        }
 
-                if (GreaseQuicBitEnabled) {
-                    TEST_TRUE(Client.GetStatistics().GreaseBitNegotiated);
-                    TEST_TRUE(Server->GetStatistics().GreaseBitNegotiated);
-                }
+        TEST_EQUAL(
+            Server->GetPeerBidiStreamCount(),
+            Client.GetLocalBidiStreamCount());
 
-                if (RandomLossPercentage == 0) {
-                    //
-                    // Don't worry about graceful shutdown if we have random
-                    // loss. It will likely just result in the maximum wait
-                    // timeout, causing the test to run longer.
-                    //
-                    Client.Shutdown(QUIC_CONNECTION_SHUTDOWN_FLAG_NONE, QUIC_TEST_NO_ERROR);
-                    if (!Client.WaitForShutdownComplete()) {
-                        return;
-                    }
+        if (GreaseQuicBitEnabled) {
+            TEST_TRUE(Client.GetStatistics().GreaseBitNegotiated);
+            TEST_TRUE(Server->GetStatistics().GreaseBitNegotiated);
+        }
 
-                    TEST_FALSE(Client.GetPeerClosed());
-                    TEST_FALSE(Client.GetTransportClosed());
-                }
+        if (RandomLossPercentage == 0) {
+            //
+            // Don't worry about graceful shutdown if we have random
+            // loss. It will likely just result in the maximum wait
+            // timeout, causing the test to run longer.
+            //
+            Client.Shutdown(QUIC_CONNECTION_SHUTDOWN_FLAG_NONE, QUIC_TEST_NO_ERROR);
+            if (!Client.WaitForShutdownComplete()) {
+                return;
             }
 
-            if (RandomLossPercentage == 0) {
-                TEST_TRUE(Server->GetPeerClosed());
-                TEST_EQUAL(Server->GetPeerCloseErrorCode(), QUIC_TEST_NO_ERROR);
-            } else {
-                Server->Shutdown(QUIC_CONNECTION_SHUTDOWN_FLAG_SILENT, 0);
-            }
+            TEST_FALSE(Client.GetPeerClosed());
+            TEST_FALSE(Client.GetTransportClosed());
         }
     }
+
+    if (RandomLossPercentage == 0) {
+        Server->WaitForPeerClose();
```

Comment:
Github shows this diff horribly.
This is the only line added, everything else is removing some indentation level and braces (the "hide whitespace" option makes it better)